File: example/magnitude.ts
let model = {
    "open" : false,
    "heading" : 0,
}

basic.forever(function on_forever() {
    input.onButtonPressed(Button.A, function onPressButton_A() {
        
        model["open"] = false
    })
    input.onButtonPressed(Button.B, function onPressButton_B() {
        
        model["open"] = true
    })
    model["heading"] = input.compassHeading()
    function viewCompass() {
        
        if (model["open"] == true) {
            basic.showNumber(model["heading"])
        } else {
            basic.clearScreen()
        }
        
    }
    
    viewCompass()
})

File: example/magnitude.py
model = {'open':False, 'heading':0}

def on_forever():
    def onPressButton_A():
        global model
        model['open'] = False
    input.on_button_pressed(Button.A, onPressButton_A)
    
    def onPressButton_B():
        global model
        model['open'] = True
    input.on_button_pressed(Button.B, onPressButton_B)
    
    model['heading'] = input.compass_heading()
    
    def viewCompass():
        global model
        if model['open'] == True:
            basic.show_number(model['heading'])
        else:
            basic.clear_screen()
    viewCompass()

basic.forever(on_forever)
